docs/src/pages/customization/styled/styled.md

```diff
@@ -0,0 +1,80 @@
+- support for [the `sx` prop](/system/basics/#the-sx-prop)
+
+### What problems does it solve?
+
+The utility can be used as a replacement for emotion's/styled-components' styled() utility.
+It aims to solve the same problem, but also provides the following benefits:
+
+1. It uses the default Material-UI's `theme` if no theme is available in the context
+2. It supports the theme's `styleOverrides` and `variants` to be applied, based on the `name` applied in the options (can be skipped)
+3. It adds support for the `sx` prop automatically (can be skipped)
+4. It adds by default `shouldForwardProp` option that is taking into account all props used internally in the Material-UI components (can be overridden)
+
+## API
+
+### `experimentalStyled(Component, [options], [muiOptions])(styles) => Component`
+
+#### Arguments
+
+1. `Component`: The component that will be wrapped.
+2. `options` (_object_ [optional]):
+
+   - `options.shouldForwardProp` (_`(props: string) => boolean`_ [optional]): Indicates whether the `prop` should be forwared to the `Component`.
+   - `options.label` (_string_ [optional]): The suffix of the style sheet. Useful for debugging.
+   - The other keys are forwarded to the options argument of emotion's [styled([Component], [options])](https://emotion.sh/docs/styled).
+
+3. `muiOptions` (_object_ [optional]):
+
+   - `muiOptions.name` (_string_ [optional]): The key used under `theme.components` for specifying `styleOverrides` and `variants`. Used also for generating the `label`.
+   - `muiOptions.slot` (_string_ [optional]): If `Root`, it automatically applies the theme's `styleOverrides` & `variants`
+   - `muiOptions.overridesResolver` (_(props: object, styles: Record<string, styles>) => styles_ [optional]): Function that returns styles based on the props and the `theme.components[name]styleOverrides` object.
+   - `muiOptions.skipVariantsResolver` (_boolean_): Disables the automatic resolver for the `theme.components[name].variants`
+   - `muiOptions.skipSx` (_bool_ [optional]): Disables the `sx` prop on the component.
+
+#### Returns
+
+`Component`: The new component created.
+
+## Basic usage
+
+{{"demo": "pages/customization/styled/BasicUsage.js", "defaultCodeOpen": true}}
+
+## Options
+
+{{"demo": "pages/customization/styled/UsingOptions.js", "defaultCodeOpen": true}}
+
+If you inspect this element with the dev tools, you will notice that the class of the compoent now ends with the `MyTestComponent`, which is the label that we provided. In addition to this, the `color` prop is not propagated to the generated `div` element.
+
+<img src="/static/images/customization/styled-options.png" alt="dev-tools" width="327" />
+
+## `muiOptions`
+
+{{"demo": "pages/customization/styled/UsingMuiOptions.js"}}
+
+If you're using TypeScript, you'll need to specify your new component's overrides/variants, using [module augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation).
```

Comment:
An example would help here.

Reply:
I've updated the example to showcase this and removed this section.